tools/mo/openvino/tools/mo/main.py

```diff
@@ -308,25 +322,53 @@ def raise_ie_not_found():
         from openvino.tools.mo.front.onnx.register_custom_ops import get_front_classes
         import_extensions.load_dirs(argv.framework, extensions, get_front_classes)
 
-    return argv
 
+def check_fallback(argv : argparse.Namespace):
+    fallback_reasons = {}
 
-def prepare_ir(argv):
-    argv = arguments_post_parsing(argv)
+    if not any(deduce_framework_by_namespace(argv)): # no legacy path
+        return fallback_reasons
+
+    if argv.use_new_frontend or argv.use_legacy_frontend: # frontend chosen explicitly
+        return fallback_reasons
+
+    fallback_reasons['extensions'] = \
+        lambda argv : hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
+            and argv.extensions != import_extensions.default_path() # extensions arg has default value
+    fallback_reasons['transformations_config'] = \
+        lambda argv: hasattr(argv, 'transformations_config') and argv.transformations_config is not None and len(argv.transformations_config) > 0
```

Comment:
Does it mean that I cannot pass JsonConfigExtension? @itikhono please look at that.

Reply:
@slyalin I guess we don't have any implemented transformations in c++ part to run via JsonConfig, doesn't that mean we should always use a fallback? 
And I think we already don't have time to implement such transformations in 2022.1R

Reply:
Discussed with @itikhono. JsonConfigExtension is a legal replacement of old config-enabled way in MO but for new FEs. It seems to be tested enough to enable it here. Regardless of fact that we don't have our own implemented JsonConfigExtension's. We don't really want to have them inside the product and leaving this way for advanced users as an extreme measure. @mbencer, please put more fine-grain condition here to let new transformation config pass to new FEs. New config can be detected by identifying a field `library` in json file at the level of each transformation. Please consult with @itikhono for details. I would prefer to have this in a separate PR to avoid blocking of validation the fallback itself. But we must have that. Otherwise all stuff made for JsonConfigExtension cannot be activated by a user.